portfolio/src/main/java/com/google/sps/servlets/UsernameManager.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static com.google.sps.data.ConstantValues.DEFAULT_VAL;
+import static com.google.sps.data.ConstantValues.INDEX_PATH;
+import static com.google.sps.data.ConstantValues.JSON_CONTENT_TYPE;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.LoginStatus;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
+
+@WebServlet("/login")
+public class UsernameManager extends HttpServlet {
+
+  private static final String USER_INFO = "UserInfo";
+  private static final String USER_ID = "id";
+  private static final String USER_NAME = "username";
+  // The object connected to datastore.
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  // The object which does user login/logout.
+  private static final UserService userService = UserServiceFactory.getUserService();
+  // The Java to JSON converter.
+  private static final Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType(JSON_CONTENT_TYPE);
+    PrintWriter out = response.getWriter();
+    // If the user has logged in, get the pre-registered username,
+    // if not, get the login url.
+    Optional<String> loginUrl = userService.isUserLoggedIn() 
+                      ? Optional.empty()
+                      : Optional.of(userService.createLoginURL(INDEX_PATH));
+    Optional<String> username = userService.isUserLoggedIn()
+                              ? Optional.of(getUsername(userService.getCurrentUser().getUserId()))
+                              : Optional.empty();
```

Comment:
Can be simplified to `Optional.of(...).` Please read the docs.

Reply:
Thanks for the reminder. I checked the docs and the lines above can be simplified to `Optional.ofNullable`